app_logic.py

```diff
@@ -0,0 +1,107 @@
+
+def user_can_make(recipe, ingredients, can_miss=0): #TODO: ingredients and can_miss global?
+    """Takes a list drink recipe and compares it to a list
+    of ingredients the user has on hand. 
+    Accepts an additional parameter of how many items can be
+    missing from the users ingredients, default to 0
+    
+    Returns True or False"""
+    misses = 0
+    for i in range(1,16):
+        if recipe[f'strIngredient{i}']:
+            if recipe[f'strIngredient{i}'] not in ingredients:
+                miss += 1
+                if miss > can_miss:
+                    return False
+        else:
+            return True
+
+
+def fetch_responses(response, seen_list, num_to_return=10): #TODO: make seen_list global?
+    """Takes a response object from the API and a 
+    list of recipe ids that have already been checked. 
+    Also can accept a paramter of how many makeable 
+    recipes to return, default to 10
+    
+    Returns a list of drink objects (of length 
+    num_to_return) of drinks user could create"""
+
+    results = 0
+    result_list = {}
+    for drink in response['drinks']:
+        if drink['idDrink'] not in seen_list:
+            recipe = getRecipe(drink['idDrink'])
+            if user_can_make(recipe):
+                result_list[drink['idDrink']] = recipe
+                seen_list.append(drink['idDrink'])
+                results += 1
+                if results == num_to_return:
+                    return result_list
+            else:
+                seen_list.append(drink['idDrink'])
+
+
+#psuedo-code
+'''
+for ingredient in list:
+    query ingredient from ingredients.name
+    compile list of ingredient ids
+
+logic type #1:
+#this code can be used with a list of 1 id (general search)
+for id in ingredient ids:
+    query recipe ids with ingredient in recipe 
+    compile containing list
+
+use count(?) to compile a weighted list 
+(how many times an id appears will mean how many ingredients listed it contains)
+
+logic type #2:
+#Again this section of code can be used with a list of 1
+for id in ingredient ids:
+    query recipe ids with ingredient in recipe
+    compile containing set
+
+for id in set:
+    pull full recipe
+    create dict for recipe
+    include item how many ingredients match
+'''
+
+
+def get_recipe_ids_with_ingredient(ingredient):
```

Comment:
Why is there a function with an identical name in another file? Is this one used? If not delete it.

https://github.com/cwverica/quixr/pull/1/files#diff-568470d013cd12e4f388206520da39ab9a4e4c3c6b95846cbc281abc1ba3c959R235

Reply:
This whole file is probably going to get deleted. I was using it before I decided to front-load the app, and it just kinda became a scratch-pad.